src/main/java/org/kohsuke/github/GHTreeBuilder.java

```diff
@@ -162,6 +171,21 @@ public GHTreeBuilder add(String path, String content, boolean executable) {
         return add(path, content.getBytes(StandardCharsets.UTF_8), executable);
     }
 
+    /**
+     * Removes an entry with the given path from base tree.
+     *
+     * @param path
+     *            the file path in the tree
+     * @param executable
+     *            true, if the file should be executable
+     * @return this GHTreeBuilder
+     */
+    public GHTreeBuilder delete(String path, boolean executable) {
+        TreeEntry entry = new DeleteTreeEntry(path, executable ? "100755" : "100644", "blob");
```

Comment:
Is `executable` required for deletion?  Does it need to match the tree object that is being deleted?

Reply:
I also wonder what happens if the `type` doesn't match the item at that `path`. 
https://docs.github.com/en/rest/git/trees?apiVersion=2022-11-28#create-a-tree under `properties of tree`, shows type can be `blob`, `tree`, or `commit`.

This isn't necessarily a blocker, but it would be good to know if there are scenarios that the current `delete()` method doesn't handle.

Reply:
> Is `executable` required for deletion? Does it need to match the tree object that is being deleted? If not (if delete works regardless of this value), we should remove the `executable` parameter here.

Yes, it's required for deletion as well and it should be not null.
Reference doc says nothing about what value for `mode` should be used for deletion but my test shows it works with any valid non-null value. So we can use some hadcoded value. I will change this

Reply:
> I also wonder what happens if the `type` doesn't match the item at that `path`.

I tested `type=commit` while deleting a `blob` and it works fine.